airflow/contrib/operators/gcp_automl_operator.py

```diff
@@ -0,0 +1,992 @@
+            input_config=self.input_config,
+            output_config=self.output_config,
+            project_id=self.project_id,
+            location=self.location,
+            params=self.params,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+        result = MessageToDict(operation.result())
+        self.log.info("Batch prediction ready.")
+        return result
+
+
+class AutoMLCreateDatasetOperator(BaseOperator):
+    """
+    Creates a Google Cloud AutoML dataset_id.
+
+    :param dataset: The dataset to create. If a dict is provided, it must be of the
+        same form as the protobuf message Dataset.
+    :type dataset: Union[dict, Dataset]
+    :param project_id: ID of the Google Cloud project where dataset is located if None then
+        default project_id is used.
+    :type project_id: str
+    :param location: The location of the project.
+    :type location: str
+    :param params: Additional domain-specific parameters for the predictions.
+    :type params: Optional[Dict[str, str]]
+    :param retry: A retry object used to retry requests. If `None` is specified, requests will not be
+        retried.
+    :type retry: Optional[google.api_core.retry.Retry]
+    :param timeout: The amount of time, in seconds, to wait for the request to complete. Note that if
+        `retry` is specified, the timeout applies to each individual attempt.
+    :type timeout: Optional[float]
+    :param metadata: Additional metadata that is provided to the method.
+    :type metadata: Optional[Sequence[Tuple[str, str]]]
+    :param gcp_conn_id: The connection ID to use to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+    """
+
+    template_fields = ("dataset", "location", "project_id")
+
+    @apply_defaults
+    def __init__(
+        self,
+        dataset: dict,
+        location: str,
+        project_id: str = None,
+        metadata: Sequence[Tuple[str, str]] = None,
+        timeout: float = None,
+        retry: Retry = None,
+        gcp_conn_id: str = "google_cloud_default",
+        *args,
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+
+        self.dataset = dataset
+        self.location = location
+        self.project_id = project_id
+        self.metadata = metadata
+        self.timeout = timeout
+        self.retry = retry
+        self.gcp_conn_id = gcp_conn_id
+
+    def execute(self, context):
+        hook = AutoMLHook(gcp_conn_id=self.gcp_conn_id)
+        self.log.info("Creating dataset")
+        result = hook.create_dataset(
+            dataset=self.dataset,
+            location=self.location,
+            project_id=self.project_id,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+        result = MessageToDict(result)
+        self.log.info("Creating completed.")
+
+        self.xcom_push(context, key="dataset_id", value=get_object_id(result))
```

Comment:
I found a similar request in Jira. What do you think about adding this to the recommendation?

https://issues.apache.org/jira/browse/AIRFLOW-3209

Reply:
I agree that this could be part of retrofitting.